mobile/src/main/java/com/gelakinetic/mtgfam/helpers/database/CardDbAdapter.java

```diff
@@ -1249,12 +1249,15 @@ public static Cursor Search(SearchCriteria criteria, boolean backface, String[]
                     + " " + criteria.cmc + ")";
         }
 
-        if (criteria.noTokens) {
+        if (criteria.moJhoStoFilter) {
+            /* Filter out tokens. */
             statement += " AND (" +
-                        "(NOT " + DATABASE_TABLE_CARDS + "." + KEY_MANACOST + " = '') " +
-                        "OR " + DATABASE_TABLE_CARDS + "." + KEY_SUPERTYPE + " LIKE '%Land%' " +
-                        "OR " + DATABASE_TABLE_CARDS + "." + KEY_ABILITY + " LIKE '%Suspend%' " +
-                        "OR " + DATABASE_TABLE_CARDS + "." + KEY_ABILITY + " LIKE '%Splice onto Arcane%')";
+                    /* Cards without mana costs. */
+                    "NOT " + DATABASE_TABLE_CARDS + "." + KEY_MANACOST + " = '' " +
+                    /* Cards like 'Dryad Arbor'. */
+                    "OR " + DATABASE_TABLE_CARDS + "." + KEY_SUPERTYPE + " LIKE '%Land Creature%')";
```

Comment:
This filter is for MoJhoSto only, so it does not need to be as generic/complicated.